core/cont/src/TClassTable.cxx

```diff
@@ -420,6 +475,7 @@ void TClassTable::Add(TProtoClass *proto)
                    // files loaded by the current dictionary wil also de-activate the update
                    // class info mechanism!
 
+         lock.unlock(); // Warning migth recursively call TClassTable during gROOT init
```

Comment:
The `proto->FillTClass()` call in line 468 can emit debug and error messages. Doesn't it have the same problem?

Reply:
You are correct.  It is a problem.  I am adding the unlock before its call.